### DISCUSSION TITLE
Restoring Qt4 support: advice needed

### DISCUSSION THREAD

**Discussion by barracuda156** [transmission/transmission#6862] [Q&A]
### What is the issue?

@ckerr Since you have dropped it earlier, could you guide me a bit on restoring?

To begin with, a general question: how tightly connected is `qt` backend (whatever in `qt` folder) to the rest of the sources? I would assume that simply replacing it with the version from 2.94 (which is confirmed to build with Qt4 for PowerPC) won’t work, but is it doable – and meaningful to forward-port the older Qt backend rather than trying to fix the current one?

If working from the current Qt backend, it would be very helpful to know which changes were incompatible with Qt4 – and which are orthogonal to Qt version. I.e., I do not want to revert improvements which are fine with Qt4.

Trying to fix `transmission` 3.00, I found these commits breaking:

caf3b7653c6125f1c99ec01a106f827316f945a0 (prior to this `transmission` builds with Qt4 from the master branch, but it reports version 2.93+, i.e. it is older than released 2.94)
8dc6fb48b1640c063fe86af5c37a21139321ffb3 (apparently the reason for it was the previous commit, though instead it could have been done conditionally)
03034b501563a9900f67ce4cb68b9535c629cfbf (obviously)
abac811dd24fd188681116f8964d17075d2c83db (ironically, it seems to break GCC build instead of fixing it)

Unfortunately, these commits do not revert from 3.00 release (forget 4.x). I began reverting select commits manually, but there are a lot, and after spending a several hours yesterday – though I got close to building 3.00 with Qt4 – I am wondering if I revert something unnecessarily.

### Which application of Transmission?

Qt app on Linux, BSD, etc.

### Which version of Transmission?

4.x, 3.x

**Comment by killemov** [comment]
Of course you can revert these specific changes with git. Looking at the code in the aforementioned commits, I can't detect the use of QT5-specific components and it generally looks like a cleanup to just get rid of QT4.

Transmission 4+ underwent a transformation to C++ and I guesstimate that your time would be better spent folding QT4 back in the recent version. Ultimately a single UI-target identifier (WEB, QT4, QT5, LEOPARD, SNOW_LEOPARD,  ... ) would be the most logical, but then again the macOS client was developed completely separately.

**Comment by barracuda156** [comment]
Might not be worth the pain, since we got the latest Transmission running with GTK GUI on PowerPC.

(Not closing for now, since Qt interfaces are generally preferable and more stable, when they can build.)
